### PR TITLE
Add scopes and selected scope view

### PR DESCRIPTION
**Pull request by helfi92**
Resolves https://github.com/taskcluster/taskcluster-web/issues/37.
Resolves https://github.com/taskcluster/taskcluster-web/issues/38.

Scopes view: A list of scopes.
Selected scope view: A list of role and client IDs.

### CODE REVIEW

**Comment by eliperelman** [comment]
Can you post some screenshots of this?

**Comment by helfi92** [comment]
Definitely. I took screenshots but forgot to upload them lol.

**Comment by helfi92** [comment]
## Scopes view

screen shot 2018-05-14 at 8 41 32 am

## Selected scope view

screen shot 2018-05-14 at 11 06 33 am

screen shot 2018-05-14 at 11 06 49 am

File: src/components/DateChooser/index.jsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import MuiPickersUtilsProvider from 'material-ui-pickers/utils/MuiPickersUtilsProvider';
+import ChevronRightIcon from 'mdi-react/ChevronRightIcon';
+import CalendarIcon from 'mdi-react/CalendarIcon';
+import ChevronLeftIcon from 'mdi-react/ChevronLeftIcon';
+import { date } from '../../utils/prop-types';
+
+/**
+ * Display a date picker modal to select a date.
+ */
+export default class DateChooser extends Component {
```
Comment (eliperelman) [inline src/components/DateChooser/index.jsx]: This class seems familiar. Didn't I cover this in another review where this was named `DatePicker`?

File: src/utils/scopeMatch.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  const hasScope = cond([
+    [exact, T],
+    [
+      test(/\\*$/),
+      pipe(dropLast(1), scope => selectedScope.indexOf(scope), equals(0)),
+    ],
+    [T, F],
+  ]);
+
+  return cond([
```
Comment (eliperelman) [inline src/utils/scopeMatch.js]: I like FP, but I can't read this function. If you have to mix a lot of arrow functions into a `cond`, you're losing the benefit of point-free syntax.

I'd probably recommend mixing some more imperative syntax in here if you can't make this fully point-free.

File: src/utils/scopeMatch.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+          [equals(pattern), T],
+          [T, scope => scope.indexOf(dropLast(1, pattern)), equals(0)],
+        ]);
+      },
+    ],
+    [T, T],
+  ]);
+};
+
+export default scopeMatch;
```
Comment (eliperelman) [inline src/utils/scopeMatch.js]: You can `export default (mode, selectedScope) => {`.

File: src/views/Scopes/index.jsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  render() {
+    const {
+      match: { path },
+      ...props
+    } = this.props;
+
+    return (
+      <Switch>
+        <RouteWithProps
+          path={`${path}/:view(clients|roles)/:selectedScope/:selectedScopeView(clients|roles)?`}
```
Comment (eliperelman) [inline src/views/Scopes/index.jsx]: Am I correct in reading that the URL is `/auth/scopes/roles/:scope/roles`? Seems kinda redundant. What if we did something like: `/auth/clients/:scope` and `/auth/roles/:scope`? There may be another variation too.

File: src/utils/scopeMatch.js (hunk 2 above)
Comment (helfi92) [reply src/utils/scopeMatch.js]: Agreed.

File: src/views/Scopes/index.jsx (hunk 4 above)
Comment (helfi92) [reply src/views/Scopes/index.jsx]: `/auth/clients/:scope` and `/auth/roles/:scope` are being used by the client and role views.

Comment (helfi92) [reply src/views/Scopes/index.jsx]: The tab layout makes URLs ugly.

Comment (eliperelman) [reply src/views/Scopes/index.jsx]: We could skip putting the tabs in the URL, and just make the default tab show up.

Comment (helfi92) [reply src/views/Scopes/index.jsx]: Works for me.

**Comment by helfi92** [comment]
## Scopes View

screen shot 2018-05-15 at 3 50 36 pm

## Selected Scope View

screen shot 2018-05-15 at 3 48 49 pm

**Pull request merged**